### PR TITLE
Feature/5279 openbaarmaken onderschrift attachment

### PR DESCRIPTION
**Pull request by OscarBakker**
Ticket: [SIG-5279](https://gemeente-amsterdam.atlassian.net/browse/SIG-5279)

## Signalen

Before opening a pull request, please ensure:

- Make sure your PR title follows naming conventions: [feat-1234]: name feature
- Double-check your branch is based on `main` and targets `main`
- Pull request has tests (we are going for 100% coverage!)
- Code is well-commented, linted and follows project conventions
- Committed source code is headed by the correct SPDX license expression

Be kind to code reviewers, please try to keep pull requests as small and focused as possible :)

### CODE REVIEW

**Review by SireeKoolenWijkstra — CHANGES_REQUESTED** [review]
~~[ ] In the attachmentviewer in the backoffice the caption needs to be visible.~~

~~- [ ] If the image is uploaded by the melder, in the attachmentviewer in the corner left above the text melder needs to be shown. Otherwise openbaar getoond (if the vinkje is checked for that)~~

**Review by SireeKoolenWijkstra — APPROVED** [review]
Looks good to me

File: src/signals/incident-management/containers/IncidentDetail/components/Attachments/EditAttachment..tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+          // @ts-ignore
+          acc[key] = data[key]
+        }
+        return acc
+      }, {} as FormValues)
+      if (!data.public && dirtyFields.public && dirtyData.caption) {
+        dirtyData = { ...dirtyData, caption: null } as FormValues & {
+          caption: null
+        }
+      }
```
Comment (vdegraaf) [inline src/signals/incident-management/containers/IncidentDetail/components/Attachments/EditAttachment..tsx:69]: What's happening here?

**Review by vdegraaf — COMMENTED** [review]
nice! Some comments:

- Paddings of the edit box are not as designed. A bit too much.
- I can exceed the maximum characters. Should be limited. 
- testing of these components is below threshold, would be nice to fix that. it was 100%.
- shouldn't there be a checkbox below this attachment?
Screenshot 2023-09-07 at 11 52 54

File: src/signals/incident-management/containers/IncidentDetail/components/Attachments/EditAttachment..tsx (hunk 1 above)
Comment (OscarBakker) [reply src/signals/incident-management/containers/IncidentDetail/components/Attachments/EditAttachment..tsx:69]: the null value can't be added through the form. This is a bit like patching the data before sending the PATH request.

**Comment by OscarBakker** [comment]
> nice! Some comments:
> 
> * Paddings of the edit box are not as designed. A bit too much.

**they are the like the other forms within Incident Detail. All the gaps within form elements are 8 (or should be)**

> * I can exceed the maximum characters. Should be limited.

**linda says we should be able to exceed. So keep the validation error**

> * testing of these components is below threshold, would be nice to fix that. it was 100%.

- [ ] increase coverage 
> * shouldn't there be a checkbox below this attachment?
> 
> Screenshot 2023-09-07 at 11 52 54

File: src/signals/incident-management/containers/IncidentDetail/components/Attachments/Attachments.tsx [hunk 2]
```diff
@@ -60,6 +62,7 @@ interface AttachmentsProps {
   className?: string
   add: (file: File) => void
   remove: (attachment: Attachment) => void
+  patch: (href: string, attachment: any) => void
```
[the remark this answers is not in the document]
Comment (OscarBakker) [reply src/signals/incident-management/containers/IncidentDetail/components/Attachments/Attachments.tsx:65]: changed it

**Review by vdegraaf — COMMENTED** [review]
If I add an attachment, put it to "openbaar", but then want to change it again, I get the screenshot below.

Screenshot 2023-09-07 at 15 43 07

**Review by vdegraaf — APPROVED** [review]
LGTM

**Pull request merged**